src/main/java/io/confluent/connect/hdfs/wal/WALFile.java

```diff
@@ -208,20 +212,29 @@ public static Option blockSize(long value) {
         long blockSize = blockSizeOption == null ? fs.getDefaultBlockSize(p) :
                          blockSizeOption.getValue();
 
-        if (appendIfExistsOption != null && appendIfExistsOption.getValue()
-            && fs.exists(p)) {
-          // Read the file and verify header details
-          try (WALFile.Reader reader =
-                   new WALFile.Reader(conf, WALFile.Reader.file(p), new Reader.OnlyHeaderOption())){
-            if (reader.getVersion() != VERSION[3]) {
-              throw new VersionMismatchException(VERSION[3], reader.getVersion());
+        try {
+          out = fs.create(p, false, bufferSize, replication, blockSize);
+        } catch (FileAlreadyExistsException e) {
+          if (appendIfExistsOption != null && appendIfExistsOption.getValue() && fs.exists(p)) {
+            out = fs.append(p, bufferSize);
+            // Read the file and verify header details
+            try (WALFile.Reader reader = new WALFile.Reader(conf, WALFile.Reader.file(p), new Reader.OnlyHeaderOption())) {
+              if (reader.getVersion() != VERSION[3]) {
+                throw new VersionMismatchException(VERSION[3], reader.getVersion());
+              }
+              sync = reader.getSync();
+              this.appendMode = true;
+            } catch (EOFException eof) {
+              FileStatus status = fs.getFileStatus(p);
+              if (status.getLen() == 0) {
+                this.appendMode = false;
+              } else {
+                throw eof;
```

Comment:
So for now we're assuming that any incomplete header that isn't an empty file will cause the connector to stop running? Is there a reason an empty file should be treated differently from an incomplete header? Wouldn't the latter imply the same problem as the former, i.e. that the previous process writing the WAL had crashed and left it in a corrupt state where it needs to be truncated/deleted to discard it?